File: Java/Minimum_Size_Subarray_Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int k = 0;
        int sum = 0;
        int min_length = Integer.MAX_VALUE;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];

            while (sum >= target) {
                min_length = Math.min(min_length, i - k + 1);
                sum -= nums[k];
                k++;
            }
        }
        return (min_length == Integer.MAX_VALUE) ? 0 : min_length;
    }
}

File: C#/Minimum_Size_Subarray_Sum.cs
public class Solution {
    public int MinSubArrayLen(int target, int[] nums) {
        int minLength = int.MaxValue;
        int currentSum = 0;
        int left = 0;
        
        for (int right = 0; right < nums.Length; right++) {
            currentSum += nums[right];
            
            while (currentSum >= target) {
                minLength = System.Math.Min(minLength, right - left + 1);
                currentSum -= nums[left];
                left++;
            }
        }
        
        return minLength == int.MaxValue ? 0 : minLength;
    }
}